cachi2/core/package_managers/bundler/main.py

```diff
@@ -26,3 +31,30 @@ def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) ->
     """Process a request for a single bundler package."""
     parse_lockfile(package_dir)
     return []
+
+
+def _prepare_for_hermetic_build(output_dir: RootedPath) -> None:
+    """Prepare a package for hermetic build by injecting necessary config."""
+    bundle_config = output_dir.join_within_root(".bundle/config").path
```

Comment:
I think we don't want to have any config file in the output dir, since users can request multiple bundler packages and this might lead to collision.

Reply:
Good point, where it belongs then?

Reply:
> Good point, where it belongs then?

It belongs in the user repo, right ?

The problem within bundler is that environment variables do not have the highest precedence (unlike all other package managers). The highest precedence is the config file. I would have to check the design doc, but I think we can set the variables only if there is no config file, and if it is, we should overwrite those settings.

There is `$BUNDLE_APP_CONFIG` variable that can overwrite default config file path :exploding_head: 
https://bundler.io/v2.5/man/bundle-config.1.html#DESCRIPTION

Reply:
> There is $BUNDLE_APP_CONFIG variable that can overwrite default config file path 🤯

We knew about that variable, but we only define a single set of variables that are global to the whole build env, so in a multi-package repo setup you'd be applying configuration from package A + our overrides to package B.

Reply:
> Good point, where does it belong then?

Have a look at https://github.com/containerbuildsystem/cachi2/blob/3398f3de3c8ce6370be06a49701645f8edffe8e3/cachi2/core/package_managers/rpm/main.py#L230, we'll have to come up with a similar (or unified?) way to dump the options to the build config. Because of multi-package repo setups I'd suggest to namespace each set of config vars by the package name we're processing, so that when issuing the `inject-files` CLI command we'll know where to dump each of the config files and write to the input user repos.